app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneViewModel.kt

```diff
@@ -0,0 +1,132 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.main.folder
+
+import android.net.Uri
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.infomaniak.lib.core.utils.SingleLiveEvent
+import com.infomaniak.mail.data.cache.mailboxContent.DraftController
+import com.infomaniak.mail.data.models.Attachment
+import com.infomaniak.mail.data.models.correspondent.Recipient
+import com.infomaniak.mail.data.models.draft.Draft.DraftMode
+import com.infomaniak.mail.data.models.message.Message
+import com.infomaniak.mail.data.models.thread.Thread
+import com.infomaniak.mail.di.IoDispatcher
+import com.infomaniak.mail.ui.main.thread.DetailedContactBottomSheetDialogArgs
+import com.infomaniak.mail.ui.main.thread.actions.DownloadAttachmentProgressDialogArgs
+import com.infomaniak.mail.ui.main.thread.actions.MessageActionsBottomSheetDialogArgs
+import com.infomaniak.mail.ui.main.thread.actions.ReplyBottomSheetDialogArgs
+import com.infomaniak.mail.ui.main.thread.actions.ThreadActionsBottomSheetDialogArgs
+import com.infomaniak.mail.ui.newMessage.NewMessageActivityArgs
+import com.infomaniak.mail.utils.Utils.runCatchingRealm
+import com.infomaniak.mail.utils.coroutineContext
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import javax.inject.Inject
+
+@HiltViewModel
+class TwoPaneViewModel @Inject constructor(
+    private val draftController: DraftController,
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
+) : ViewModel() {
+
+    private val ioCoroutineContext = viewModelScope.coroutineContext(ioDispatcher)
+
+    val currentThreadUid = MutableLiveData<String?>(null)
+
+    inline val isThreadOpen get() = currentThreadUid.value != null
+    val rightPaneFolderName = MutableLiveData<String>()
+    var previousFolderId: String? = null
+
+    val downloadAttachmentArgs = SingleLiveEvent<DownloadAttachmentProgressDialogArgs>()
+    val newMessageArgs = SingleLiveEvent<NewMessageActivityArgs>()
+    val replyBottomSheetArgs = SingleLiveEvent<ReplyBottomSheetDialogArgs>()
+    val threadActionsArgs = SingleLiveEvent<ThreadActionsBottomSheetDialogArgs>()
+    val messageActionsArgs = SingleLiveEvent<MessageActionsBottomSheetDialogArgs>()
+    val detailedContactArgs = SingleLiveEvent<DetailedContactBottomSheetDialogArgs>()
+
+    fun openThread(uid: String) {
+        currentThreadUid.value = uid
+    }
+
+    fun closeThread() {
+        currentThreadUid.value = null
+    }
+
+    fun openDraft(thread: Thread) {
+        navigateToSelectedDraft(thread.messages.first())
```

Comment:
Can we use `.single()` instead of `.first()` ?

Reply:
Yes we can.